daml-model/daml/Trigger/LoanManagement.daml

```diff
@@ -0,0 +1,53 @@
+--
+-- Copyright (c) 2022, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+--
+
+module Trigger.LoanManagement where
+
+import DA.Action (void)
+import DA.Foldable (forA_)
+import DA.Optional
+import Daml.Trigger
+import Model.Balance
+import Model.LoanConfig
+
+
+loanManagementTrigger : Trigger ()
+loanManagementTrigger = Trigger
+    { initialize = pure ()
+    , updateState = \_ -> pure ()
+    , rule = createLoanWhenBelowThreshold
+    , registeredTemplates = RegisteredTemplates [
+        registeredTemplate @Balance,
+        registeredTemplate @LoanConfig
+        ]
+    , heartbeat = None
+    }
+
+createLoanWhenBelowThreshold : Party -> TriggerA () ()
+createLoanWhenBelowThreshold sender = do
+    loanConfigs <- query @LoanConfig
+    balances <- fmap (\x -> snd <$> x) $ query @Balance
+    forA_ loanConfigs \(loanCid, loanConfig) -> do
+        let maybeBalance = listToOptional $ filter (correspondsTo loanConfig) balances
```

Comment:
Just an idea, maybe we could build a Map from the initial Balance list.
(provider, address) -> Balance

Reply:
Indeed. Converted to Map accordingly.